### PR TITLE
refactor: table deployments loader and viewer refresh button

### PR DESCRIPTION
**Pull request by J1za**
## Linear Ticket
https://linear.app/autokitteh/issue/UI-859/deployments-display-loader-only-on-first-load-dont-blink-on-following
https://linear.app/autokitteh/issue/UI-853/session-viewer-disable-refresh-if-the-status-is-finished
## What type of PR is this? (check all applicable)

-   [ ] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/components/organisms/deployments/sessions/viewer.tsx [hunk 1]
```diff
@@ -133,6 +133,11 @@ export const SessionViewer = () => {
 		};
 	}, [sessionInfo, formatTimeDifference]);
 
+	const visibleRefreshButton = useMemo(
+		() => sessionInfo?.state === SessionState.created || sessionInfo?.state === SessionState.running,
+		[sessionInfo?.state]
+	);
```
Comment (RonenMars) [inline src/components/organisms/deployments/sessions/viewer.tsx:139]: In the future, we will have "Pending" status.
So you better do this:
```suggestion
	const visibleRefreshButton = useMemo(
		() => sessionInfo?.state !== SessionState.completed || sessionInfo?.state !== SessionState.error || sessionInfo?.state !== SessionState.stopped,
		[sessionInfo?.state]
	);
```

Comment (RonenMars) [reply src/components/organisms/deployments/sessions/viewer.tsx:139]: Also only disable the button, don't hide it.

Comment (J1za) [reply src/components/organisms/deployments/sessions/viewer.tsx:139]: changed

File: src/components/organisms/deployments/sessions/viewer.tsx [hunk 2]
```diff
@@ -133,6 +133,14 @@ export const SessionViewer = () => {
 		};
 	}, [sessionInfo, formatTimeDifference]);
 
+	const visibleRefreshButton = useMemo(
```
Comment (RonenMars) [inline src/components/organisms/deployments/sessions/viewer.tsx:136]:
```suggestion
	const isRefreshButtonDisabled = useMemo(
```

**Review by RonenMars — APPROVED** [review]
Just fix the constant name

File: src/components/organisms/deployments/sessions/viewer.tsx (hunk 2 above)
Comment (J1za) [reply src/components/organisms/deployments/sessions/viewer.tsx:136]: changed

**Pull request merged**